scripts/gdb-optee.py

```diff
@@ -0,0 +1,326 @@
+
+            # FIXME: This must be updated to support QEMU v8 for example (path ...)
+            gdb.execute("set sysroot {}/{}".format(OPTEE_PROJ_PATH, "out-br/host/arm-buildroot-linux-gnueabihf/sysroot"))
+            gdb.execute("b main")
+
+        except IndexError:
+            print("No host binary specified")
+
+    def complete(self, text, word):
+        # Sync the array with invoke
+        candidates = ['hello_world', 'hotp', 'random', 'acipher', 'secure_storage', 'xtest']
+        return filter(lambda candidate: candidate.startswith(word), candidates)
+
+LoadHost()
+
+class LoadTFA(gdb.Command):
+    def __init__(self):
+        super(LoadTFA, self).__init__("load_tfa", gdb.COMMAND_USER)
+
+    def invoke(self, arg, from_tty):
+        try:
+            print("Loading symbols for Trusted Firmware A from '{}'".format(arg))
+            binary = None
+            if arg == "bl1":
+                binary = BL1_ELF
+            elif arg == "bl2":
+                binary = BL2_ELF
+            elif arg == "bl31":
+                binary = BL31_ELF
+            else:
+                print("Unknown/unspecified TF-A binary!")
+                return
+            gdb.execute("symbol-file {}/{}".format(OPTEE_PROJ_PATH, binary))
+
+            if binary == BL1_ELF:
+                gdb.execute("b bl1_entrypoint")
+                gdb.execute("b bl1_main")
+            elif binary == BL2_ELF:
+                gdb.execute("b bl2_entrypoint")
+                gdb.execute("b bl2_main")
+            elif binary == BL31_ELF:
+                gdb.execute("b bl31_entrypoint")
+                gdb.execute("b bl31_main")
+                gdb.execute("b opteed_setup")
+                gdb.execute("b opteed_init")
+                gdb.execute("b opteed_smc_handler")
+
+        except IndexError:
+            print("No TF-A binary specified")
+
+    def complete(self, text, word):
+        # Sync the array with invoke
+        candidates = ['bl1', 'bl2', 'bl31']
+        return filter(lambda candidate: candidate.startswith(word), candidates)
+
+LoadTFA()
+
+class LoadLinux(gdb.Command):
+    def __init__(self):
+        super(LoadLinux, self).__init__("load_linux", gdb.COMMAND_USER)
+
+    def invoke(self, arg, from_tty):
+        print("Loading symbols for Linux kernel")
+        gdb.execute("symbol-file {}/{}".format(OPTEE_PROJ_PATH, LINUX_KERNEL_ELF))
+        gdb.execute("b tee_init")
+        gdb.execute("b optee_do_call_with_arg")
+        gdb.execute("b optee_probe")
+
+LoadLinux()
+
+class LoadUBoot(gdb.Command):
+    def __init__(self):
+        super(LoadUBoot, self).__init__("load_uboot", gdb.COMMAND_USER)
+
+    def invoke(self, arg, from_tty):
+        print("Loading symbols for U-Boot")
+        gdb.execute("symbol-file {}/{}".format(OPTEE_PROJ_PATH, UBOOT_ELF))
+        gdb.execute("b _main")
+
+LoadUBoot()
+
+class OPTEECmd(gdb.Command):
+    def __init__(self):
+        super(OPTEECmd, self).__init__("optee-stat", gdb.COMMAND_USER)
```

Comment:
This command is a bit of "WIP", should probably not be submitted in a first official patch.